ingestion-beam/src/main/java/com/mozilla/telemetry/io/Read.java

```diff
@@ -46,8 +48,12 @@ public PubsubInput(ValueProvider<String> subscription) {
     @Override
     public Result<PCollection<PubsubMessage>> expand(PBegin input) {
       return input //
-          .apply(PubsubIO.readMessagesWithAttributes().fromSubscription(subscription))
-          .apply(ToPubsubMessageFrom.identity());
+          .apply(PubsubIO.readMessagesWithAttributesAndMessageId().fromSubscription(subscription))
+          .apply(MapElements.into(TypeDescriptor.of(PubsubMessage.class)).via(message -> {
+            Map<String, String> attributesWithMessageId = message.getAttributeMap();
+            attributesWithMessageId.put(Attribute.MESSAGE_ID, message.getMessageId());
```

Comment:
This line is mutating the `attributeMap` from the input PubsubMessage. We are in general not guaranteed that the Map implementation used in a PubsubMessage is mutable, so I would normally recommend that we populate `new HashMap<>(message.getAttributeMap())`. In practice, this transform is happening directly after `PubsubIO.readMessagesWithAttributesAndMessageId()` which returns a mutable map, so probably fine.

If we want to keep this as a mutation to save allocating a new map, it would be simpler and more explicit to call `message.getAttributeMap().put(...)` and return `message` rather than constructing a new one. Otherwise, let's copy the map.

Reply:
Okay it seems this is causing some problems: https://circleci.com/gh/mozilla/gcp-ingestion/18849?utm_campaign=workflow-failed&utm_medium=email&utm_source=notification

Will create a PR to fix this.